### DISCUSSION TITLE
Proper way to use cmdk and hotkeys together

### DISCUSSION THREAD

**Discussion by Amirthananth** [pacocoursey/cmdk#200] [Q&A]
I am trying to implement cmdk command palette along with some actions in the list to be executed on click. I also want to have hotkeys to work independently.. For instance, linear.app / intercom have this feature where we can just click the hotkey and the focus would be shifted or the intended action will be applied. The same action can be performed via command palette as well. 

Not sure how to achieve the action via command palette(cmdk) when the components are deep down in the tree.. Couldn't find a proper example on the internet as well. 

Sample cmdk code

```
<Command.Dialog open={open} onOpenChange={setOpen}>
        <div className='linear'>
          <Command>
            <div cmdk-linear-badge=''>Issue - FUN-343</div>
            <Command.Input
              autoFocus
              placeholder='Type a command or search...'
            />
            <Command.List>
              <Command.Empty>No results found.</Command.Empty>
              {items.map(({ icon, label, shortcut }) => {
                return (
                  <Command.Item key={label} value={label}>
                    {icon}
                    {label}
                    <div cmdk-linear-shortcuts=''>
                      {shortcut.map((key) => {
                        return <kbd key={key}>{key}</kbd>;
                      })}
                    </div>
                  </Command.Item>
                );
              })}
            </Command.List>
          </Command>
        </div>
      </Command.Dialog>
```

**Comment by vanessa** [comment]
Have you found a solution for this? I'm experiencing the same issue.

**Reply by Amirthananth** [reply]
Apparently cmdk can only used to only render those command boxes using a shortcut like "cmd + k". Then within those options when we pick one we can make it trigger a function. 

If you need to additionally add other hotkey shortcuts then you should use some other library like blueprintjs- https://blueprintjs.com/docs/#core/hooks/use-hotkeys
